psec/pinblock.py

```diff
@@ -164,6 +171,157 @@ def encode_pinblock_iso_3(pin: str, pan: str) -> bytes:
     return _tools.xor(pinblock, pan_block)
 
 
+def encode_pin_field_iso_4(pin: str, det_pad: Optional[bytes] = None) -> bytes:
```

Comment:
Could you make `det_pad` keyword only parameter like this:

```
def encode_pin_field_iso_4(pin: str, *, det_pad: Optional[bytes] = None) -> bytes:
```

This is to make sure the person putting this here understand the implications. Could also spell it out? E.g. `deterministic_padding` instead of `det_pad`.

Reply:
Also, I don't mind deterministic padding in general. However, I always thought that whoever needed this could always monkey patch the library. For example, let's say we use `urandom` here.

This module:

```
from os import urandom as _urandom

...
```

User that wants deterministic tests:

```
from psec import pinblock

pinblock._urandom = lambda n: b"f" * n
```

What do you think? Again, I don't mind extra parameters as realistically people who use library most likely use it to drive test data against an HSM. But there is also another way.

Reply:
I have removed the deterministic padding option. I have just left one deterministic unit test in the way you described, if that is ok for you. Because you don't know what for the library will be used, it might be better to err on the side of caution.